### DISCUSSION TITLE
queryCustomObjects usage

### DISCUSSION THREAD

**Discussion by ibaykent** [SalesforceCommerceCloud/sfcc-ci#519] [Team Posts]
Hello all,

Sorry for interrupting you but I have a tiny question. I'm trying to present questions inside of select-option to the user which will filtered with the gender value. 
For example if the gender value is 1 then 1_1 1_2 and 1_3 will be displayed in the dropdown menu.
![image](https://user-images.githubusercontent.com/46345747/51893232-5a25f300-23b5-11e9-95e5-deaf7d6efb93.png)

So above is my custom object and I need to get the object and filter it. I tried to use queryCustomObject and LIKE like below. But I am probably doing something wrong. Does anyone has idea about how can I get the objects with queryCustomObjects() ?
![image](https://user-images.githubusercontent.com/46345747/51893364-b0933180-23b5-11e9-85d4-5947bdedfb6d.png)

With all my kind regards, I'm waiting your comments.

Thank you!

**Comment by zlatinz** [comment]
There are several errors in your syntax.

1. All attributes of the custom object are **custom attributes**, so you need to prefix attribute **questionid** with `custom.`
2. **String literals** in the query syntax should be surrounded by **single quotes**, as per the documentation https://documentation.b2c.commercecloud.salesforce.com/DOC1/topic/com.demandware.dochelp/DWAPI/scriptapi/html/api/class_dw_object_CustomObjectMgr.html#dw_object_CustomObjectMgr_queryCustomObject_String_String_Object_DetailAnchor
3. I am not sure what you are trying to achieve by calling the class as a function. The **queryCustomObject** method is a **static** one, so you just need class reference to be able to call it, you do not need an instance of the **CustomObjectMgr** class.

In result your code should look like
`var co = require("dw/object/CustomObjectMgr").queryCustomObjects("AvatarsPage", "custom.questionid LIKE '1_*'", null);`

Now another consideration is around queries on custom objects. It is a best practice to not use them in storefront context if you can avoid it for performance reasons. (There are no indexes on the table columns for the custom attributes and all the custom attributes are in name-value columns, so queries involving custom attributes are slower).

Can you consider why you need separate custom objects for each of the questions, can you combine them in question set (single custom object with set-of-string attribute/attributes) based on gender? This way you would request the object by its complete key value, which is the best practice for storefront context.

There are also other considerations, like do the questions change regularly, is there need for an interface to edit them, etc. which may guide you to different options for storing them, like generic mappings, Velocity properties files, JSON file, etc.

**Comment by laurent-lasgorceix-epam** [comment]
Hi,

You should remove first brackets, CustomObjectMgr is an object. I also recommend you to use the placeholder syntax to pass search parameters. Try this :

`var co = CustomObjectMgr.queryCustomObjects('AvatarsPage', 'custom.questionid LIKE {0}, '', '1_*');`
